meta-xt-control-domain/recipes-guest/doma/u-boot-android.bb

```diff
@@ -10,6 +10,8 @@ SRC_URI = "\
     ${UBOOT_SOURCE} \
 "
 
+LIC_FILES_CHKSUM = "file://Licenses/README;md5=30503fd321432fc713238f582193b78e"
```

Comment:
Hi @lorc,

It took me a while to get what you mean. I've written a big investigation. But finally got your point. So my message would be much shorter. ))

Yes. That is possible to update U-Boot and allow us not to stick to the old checksum.

But the very next change after this Kirkstone PR-s would be switching Android U-Boot to the head of Android U-Boot from the one we are using now ( patched vanilla U-boot ).

Here is its location:
https://github.com/svlad-90/u-boot/tree/xenvm-trout-dev

Here is its placeholder in xen-troops, where I'll integrate it really soon:
https://github.com/xen-troops/u-boot/tree/xenvm-trout-main

And here is what this file would look like:
https://github.com/svlad-90/meta-xt-common/blob/xenvm-trout-dev/meta-xt-control-domain/recipes-guest/doma/u-boot-android.bb

As you can see, this file will totally changed, and the U-Boot , which is currently used will be thrown away.

**=> Thus, I propose not to waste efforts on updating something, that is actually deprecated. Let's leave it as is for this PR ( as it works ). It's merge will unblock the above changes, which will make proposed update of U-Boot useless.**

Reply:
Sorry, but this is not how it works. We don't need useless patches in the codebase. If you are planning to switch to another U-Boot - just prepare the corresponding patch. 

Please take into account that somebody needs to review your changes. If you are adding a throw-away commit, this is kind of disrespect to reviewer's time.

Reply:
Hi @lorc,

This patch was done by me several months ago when I had less experience with Yocto. Yes, I didn't figure out, that I need to update U-Boot sources to a certain version. But even more experienced colleagues, who supervised me in my task, did not come to this conclusion. So, be sure, this commit was not added in order to show disrespect or uselessly spend your time.

Now ( especially after your comment ) I understand, that to get this PR merged we should go the other way around and update the U-Boot code base to a newer version. The one, which Kirkstone Yocto expects to get.

That would be the right way. I agree. But that also means another set of activities to:

- Rebase U-Boot in our branch and ( possibly ) merge conflicts with xen-troop's specific patches ( there are some dozens of those )
- Build it all
- Test it
- Pass another big review in order to get updated U-Boot merged

But I'm even not interested in having that new ( kirkstone-specific ) U-Boot, cause we already preparing changes to the product, to switch from it to Android-specific U-Boot, maintained by Google. And that U-Boot will have totally another checksum, which I will explicitly mention in this file. That is the next integration step after switch to the Kirkstone will be over.

So, from my point of view, your proposal looks like a big effort without a valuable result.

**=> Is it sufficient to update a License README in the U-Boot repo, so that we have checksum fit without the actual update of the whole U-Boot source code base? That would allow us to remove the checksum workaround. Or that also wouldn't work?**

Reply:
> Is it sufficient to update a License README in the U-Boot repo

It depends. Do you have a legal right to update License in this repo, without updating the rest of the codease? I don't know, honestly.

Anyways, I reviewed the problem more closely and I can see that it is caused by our own oversight in the first place. Basically, we are overriding `UBOOT_SOURCE` without overriding license checksum too. So, I propose the next solution. You change this line to

```
LIC_FILES_CHKSUM ??=
```

to match  `UBOOT_SOURCE ??=` above. And in the commit message you are write that you are fixing the initial problem where `u-boot-android`  recipe changed URL to the U-Boot repo, but didn't took into account that licence checksum should correspond to license file in the new URL.

Reply:
Hi @lorc,

**It seems that now we are on the same page. Thanks for taking the time to understand the context!**

----

There is only one more thing.

You said, that I should change the LIC_FILES_CHKSUM to the value, which fits the UBOOT_SOURCE.

Well, actually it is already like that.

I've downloaded this file:
https://github.com/xen-troops/u-boot/blob/android-master/Licenses/README

And calculated the md5sum:

```
vladyslav_goncharuk@XXXX:~/Projects/tmp$ md5sum ~/Downloads/README.txt 
30503fd321432fc713238f582193b78e  /home/vladyslav_goncharuk/Downloads/README.txt
```

The value **30503fd321432fc713238f582193b78e** is equal to the one in the commit:

```
LIC_FILES_CHKSUM = "file://Licenses/README;md5=30503fd321432fc713238f582193b78e"
```

**=> So I think it would be enough to change the commit message. I'll do that next week.**

Thanks, once again.

Reply:
Hi,

You missed `??=` part.

Reply:
Hi @lorc, here I have a question.

Poky Kirkstone release defines LIC_FILES_CHKSUM for U-Boot here:

https://github.com/yoctoproject/poky/blob/kirkstone/meta/recipes-bsp/u-boot/u-boot-common.inc#L10

u-boot-common.inc file is the one, which we include into the u-boot-android.bb as a first line.

Won't it lead to the fact, that "LIC_FILES_CHKSUM ??=" directive would be ignored, as it was defined earlier?

E.g. the same file does not define UBOOT_SOURCE. And UBOOT_SOURCE is used to redefine SRC_URI with a usual assignment operator - "=".

**=> Maybe we should use the same "=" operator also to override the "LIC_FILES_CHKSUM" value?**

Also, I'm not quite sure why UBOOT_SOURCE exists at all. We could assign the URI to SRC_URI directly. But ( I guess ) probably it is done to be able to override this value from the moulin yaml.

Reply:
Ah, I see. Makes sense. I don't think we need to overcomplicate things, so let's go with `=` operator.

Reply:
"+"